projectq/backends/_azure/_azure_quantum.py

```diff
@@ -0,0 +1,403 @@
+        self._circuit = None
+        self._measured_ids = []
+        self._probabilities = {}
+        self._clear = True
+
+    def _reset(self):
+        """
+        Reset this backend.
+
+        Note:
+            This sets ``_clear = True``, which will trigger state cleanup on the next call to ``_store``.
+        """
+        # Lastly, reset internal state for measured IDs and circuit body.
+        self._circuit = None
+        self._clear = True
+
+    def _store(self, cmd):
+        """
+        Temporarily store the command cmd.
+
+        Translates the command and stores it in a local variable (self._cmds).
+
+        Args:
+            cmd: Command to store
+        """
+
+        if self._clear:
+            self._probabilities = {}
+            self._clear = False
+            self._circuit = None
+
+        gate = cmd.gate
+
+        # No-op/Meta gates
+        if isinstance(gate, (AllocateQubitGate, DeallocateQubitGate)):
+            return
+
+        # Measurement
+        if isinstance(gate, MeasureGate):
+            logical_id = None
+            for tag in cmd.tags:
+                if isinstance(tag, LogicalQubitIDTag):
+                    logical_id = tag.logical_qubit_id
+                    break
+
+            if logical_id is None:
+                raise RuntimeError('No LogicalQubitIDTag found in command!')
+
+            self._measured_ids.append(logical_id)
+            return
+
+        if self._provider_id == IONQ_PROVIDER_ID:
+            if not self._circuit:
+                self._circuit = []
+
+            json_cmd = to_json(cmd)
+            if json_cmd:
+                self._circuit.append(json_cmd)
+
+        elif self._provider_id == QUANTINUUM_PROVIDER_ID:
+            if not self._circuit:
+                self._circuit = ''
+
+            qasm_cmd = to_qasm(cmd)
+            if qasm_cmd:
+                self._circuit += '\n{0}'.format(qasm_cmd)
+
+    def is_available(self, cmd):
+        """
+        Test if this backend is available to process the provided command.
+
+        Args:
+            cmd (Command): A command to process.
+
+        Returns:
+            bool: If this backend can process the command.
+        """
+        if self._provider_id == IONQ_PROVIDER_ID:
+            return is_available_ionq(cmd)
+        elif self._provider_id == QUANTINUUM_PROVIDER_ID:
+            return is_available_quantinuum(cmd)
+
+        return False
+
+    @property
+    def _target_factory(self):
+        target_factory = TargetFactory(
+            base_cls=Target,  # noqa
+            workspace=self._workspace,
+            default_targets=AzureQuantumBackend.DEFAULT_TARGETS
+        )
+
+        return target_factory
+
+    @property
+    def _target(self):
+        target = self._target_factory.get_targets(
+            name=self._target_name,
+            provider_id=self._provider_id
+        )
+
+        if type(target) is list and len(target) == 0:  # pragma: no cover
```

Comment:
Use `isinstance()` rather than `type()` to do type checks.

Reply:
Updated.